src/js/binaryen.js-post.js

```diff
@@ -2278,14 +2270,19 @@ function wrapModule(module, self) {
     return {
       'imported': Boolean(Module['_BinaryenIsFunctionTableImported'](module)),
       'segments': (function() {
-        var arr = [];
-        for (var i = 0, numSegments = Module['_BinaryenGetNumFunctionTableSegments'](module); i !== numSegments; ++i) {
-          var seg = {'offset': Module['_BinaryenGetFunctionTableSegmentOffset'](module, i), 'names': []};
-          for (var j = 0, segmentLength = Module['_BinaryenGetFunctionTableSegmentLength'](module, i); j !== segmentLength; ++j) {
+        var numSegments = Module['_BinaryenGetNumFunctionTableSegments'](module)
+        var arr = new Array(numSegments);
+        for (var i = 0; i !== numSegments; ++i) {
+          var segmentLength = Module['_BinaryenGetFunctionTableSegmentLength'](module, i);
+          var seg = {
+            'offset': Module['_BinaryenGetFunctionTableSegmentOffset'](module, i),
+            'names': new Array(segmentLength)
+          };
+          for (var j = 0; j !== segmentLength; ++j) {
             var ptr = Module['_BinaryenGetFunctionTableSegmentData'](module, i, j);
-            seg['names'].push(UTF8ToString(ptr));
+            seg['names'][j] = UTF8ToString(ptr);
```

Comment:
Maybe also cache `seg['names']`. Might be irrelevant in modern day JIT practice, but would match the overall code style here.

Reply:
Not sure about this. It increase size a little but most of JS engines should hoist such properties out of loop during jit compilation.